<a href="https://colab.research.google.com/github/mfelippetti/Machine_Learning/blob/master/Copy_of_LoteriasCaixa.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import json
import requests
import bs4

In [41]:
#site = requests.get('http://loterias.caixa.gov.br/wps/portal/loterias/landing/federal/')
site = requests.get('http://loterias.caixa.gov.br/wps/portal/loterias/landing/quina/')

In [3]:
site

<Response [200]>

In [4]:
print(site.text)

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"
	itemscope itemtype="http://schema.org/Other"
	prefix="og: http://ogp.me/ns#">
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=Edge"></meta>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
	<meta name="twitter:card" content="summary" />
	<meta property="og:type" content="website" />
	<meta property="og:title" content="Loterias Caixa" />
	<meta property="og:description" content="Acompanhe a sua sorte, Loterias Caixa, resultados, sorteios e muito mais." />
	<meta property="og:image" content="images/institucional/bolas-loterias.png" />
	<meta itemprop="name" content="Loterias Caixa"></meta>
	<meta itemprop="description" content="Acompanhe a sua sorte, Loterias Caixa, resultados, sorteios e muito mais."></meta>
	<meta itemprop="image" content="images/institucional/bolas-lote

In [42]:
soup = bs4.BeautifulSoup(site.text, 'html.parser')
print(soup)

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html itemscope="" itemtype="http://schema.org/Other" lang="en" prefix="og: http://ogp.me/ns#" xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta content="IE=Edge" http-equiv="X-UA-Compatible"/>
<meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
<meta content="summary" name="twitter:card"/>
<meta content="website" property="og:type"/>
<meta content="Loterias Caixa" property="og:title"/>
<meta content="Acompanhe a sua sorte, Loterias Caixa, resultados, sorteios e muito mais." property="og:description"/>
<meta content="images/institucional/bolas-loterias.png" property="og:image"/>
<meta content="Loterias Caixa" itemprop="name"/>
<meta content="Acompanhe a sua sorte, Loterias Caixa, resultados, sorteios e muito mais." itemprop="description"/>
<meta content="images/institucional/bolas-loterias.png" itemprop="image"/>
<meta content="width

In [43]:
base = soup.find("base")['href']
urlBuscaResultado = soup.find("input", {"id": "urlBuscarResultado", "type": "hidden"})['value']

In [44]:
resp = requests.get(base + urlBuscaResultado)
resultado = json.loads(resp.text)


In [45]:
print(resultado)

{'proximoConcurso': '5108', 'concursoAnterior': '5107', 'forward': None, 'mensagens': [], 'concurso': 5108, 'data': 1572231600000, 'resultado': '45-50-09-52-58', 'ganhadores': 0, 'ganhadores_quadra': 40, 'ganhadores_terno': 4486, 'qt_ganhador_duque': 119160, 'valor': 0.0, 'valor_quadra': 9850.71, 'valor_terno': 132.08, 'vr_rateio_duque': 2.73, 'no_cidade': 'SÃO PAULO', 'sg_uf': 'SP', 'vrAcumulado': 1354012.76, 'vrAcumuladoEspecial': 47936950.7, 'vrEstimado': 2100000.0, 'dtProximoConcurso': 1572318000000, 'vrArrecadado': 6834181.5, 'observacao': None, 'de_local_sorteio': 'Espaço Loterias Caixa', 'ic_concurso_especial': False, 'resultadoOrdenado': '09-45-50-52-58', 'duque': '0', 'error': False, 'rateioProcessamento': False, 'verificaAcumulado': False, 'sorteioAcumulado': True, 'localSorteio': 'no Espaço Loterias Caixa', 'ganhadoresPorUf': None, 'quinaSaoJoao': False, 'dataStr': '28/10/2019', 'dtProximoConcursoStr': '29/10/2019'}


In [12]:
#Resultado para QUINA
#print(resultado['resultadoOrdenado'].split('-'))

#Resultado para FEDERAL
print(resultado['premios'])

[{'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '001', 'valor': '500.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '018857', 'noMunicipioCD': 'CARANGOLA', 'noSgUfCD': 'MG', 'noFantasiaCD': 'CARANGOLA LOTERIAS'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '002', 'valor': 

In [40]:
resultado2 = resultado['premios']
resultado3 = resultado2[0]
resultado3

#resultado3['premio1']



{'proximoConcurso': '05436',
 'concursoAnterior': '05434',
 'forward': None,
 'mensagens': [],
 'extracao': '05435',
 'planoSorteio': '638',
 'dataExtracao': '26/10/2019',
 'premio1': '18857',
 'premio2': '97203',
 'premio3': '24703',
 'premio4': '50550',
 'premio5': '74380',
 'observacao': '',
 'informacao': None,
 'icLocalSorteio': '',
 'noCidade': 'SAO PAULO',
 'sgUf': 'SP',
 'faixa': '001',
 'valor': '500.000,00',
 'series': None,
 'tipoPlano': 'N',
 'co_bilhete': None,
 'qtde_bilhetes': '100000',
 'error': False,
 'bilhete': '018857',
 'noMunicipioCD': 'CARANGOLA',
 'noSgUfCD': 'MG',
 'noFantasiaCD': 'CARANGOLA LOTERIAS'}

**Resultado para FEDERAL:**
---

{'concursoAnterior': '05434', 'premios': [{'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '001', 'valor': '500.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '018857', 'noMunicipioCD': 'CARANGOLA', 'noSgUfCD': 'MG', 'noFantasiaCD': 'CARANGOLA LOTERIAS'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '002', 'valor': '27.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '097203', 'noMunicipioCD': 'SANTO ANDRE', 'noSgUfCD': 'SP', 'noFantasiaCD': 'COMPUSELECT LOTS LTD'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '003', 'valor': '24.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '024703', 'noMunicipioCD': 'CURITIBA', 'noSgUfCD': 'PR', 'noFantasiaCD': 'MONACO LOTERIAS'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '004', 'valor': '19.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '050550', 'noMunicipioCD': 'GOIANIA', 'noSgUfCD': 'GO', 'noFantasiaCD': 'LOTERICA ESTACAO DA SORTE'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '005', 'valor': '18.329,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '074380', 'noMunicipioCD': 'RIO DE JANEIRO', 'noSgUfCD': 'RJ', 'noFantasiaCD': 'QUINA LOTERICA LTDA'}], 'cidade': 'SAO PAULO', 'observacao': '', 'proximoConcurso': '05436', 'data': '26/10/2019', 'concurso': '05435', 'ufSorteio': 'SP'}



**Resultado para QUINA**
---
{'proximoConcurso': '5107', 'concursoAnterior': '5106', 'forward': None, 'mensagens': [], 'concurso': 5107, 'data': 1572058800000, 'resultado': '04-58-60-17-31', 'ganhadores': 0, 'ganhadores_quadra': 55, 'ganhadores_terno': 4382, 'qt_ganhador_duque': 111279, 'valor': 0.0, 'valor_quadra': 6200.14, 'valor_terno': 117.02, 'vr_rateio_duque': 2.53, 'no_cidade': 'SÃO PAULO', 'sg_uf': 'SP', 'vrAcumulado': 628171.43, 'vrAcumuladoEspecial': 47625875.82, 'vrEstimado': 1300000.0, 'dtProximoConcurso': 1572231600000, 'vrArrecadado': 5914567.5, 'observacao': None, 'de_local_sorteio': 'Espaço Loterias Caixa', 'ic_concurso_especial': False, 'resultadoOrdenado': '04-17-31-58-60', 'duque': '0', 'error': False, 'rateioProcessamento': False, 'verificaAcumulado': False, 'sorteioAcumulado': True, 'localSorteio': 'no Espaço Loterias Caixa', 'ganhadoresPorUf': None, 'quinaSaoJoao': False, 'dataStr': '26/10/2019', 'dtProximoConcursoStr': '28/10/2019'}

